### PR TITLE
Create Metrics

### LINKED ISSUE #314: Article: Metrics
**Article title:** *Metrics*

**Directory:** *Building and research*

**Description**: *Metrics for machine translation quality evaluation*

**Definition**: *Metrics are algorithms for evaluating machine translation outputs. Popular metrics are BLEU, NIST, METEOR, and LEPOR.*

### PR DESCRIPTION
**Pull request by cefoo**
# Create Metrics articles

Fixes #314

## Type of PR

- Creates `yisi.md`, `comet.md`, `bertscore.md`, `nist.md`, `chrf.md`, `ter.md`
- Edits `metrics.md`, `bleu.md`, `meteor.md`

### Checklist:

- [x] I have read the [contributing guidelines](contributing.md).
- [x] I have followed the [style guide](http://machinetranslate.org/style).

### CODE REVIEW

File: quality/metrics/bertscore.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+title: BERTScore
+description:
+---
+
+The **BERTScore** metric calculates the similarity in meaning between the machine translation output and a reference.
+BERTScore uses contextual embeddings.
+
+> BERTScore addresses two common pitfalls in n-gram-based metrics. [...] First, such methods often fail to robustly match paraphrases. [...] Second, n-gram models fail to capture distant dependencies and penalize semantically-critical ordering changes For example, given a small window of size two, BLEU will only mildly penalize swapping of cause and effect clauses (e.g. A because B instead of B because A), especially when the arguments A and B are long phrases. In contrast, contextualized embeddings are trained to effectively capture distant dependencies and ordering.
+
+The BERTScore metric uses BERT sentence representation.
```
Comment (bittlingmayer) [inline quality/metrics/bertscore.md:14]: > ... uses sentence representations from the BERT model.

File: quality/metrics/bertscore.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+BERTScore computes precision, recall, and F1 measure.
```
Comment (bittlingmayer) [inline quality/metrics/bertscore.md:15]: I think this can be removed.  It's more like "from BERTScore, it's possible to calculate precision, recall and F1", but that's true for any score.

File: quality/metrics/bleu.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+BLEU treats all n-grams equally.
+
+### Flavours
+
+- sacreBLEU
+- BLEUrt
+- M-BLEU
+
+*Note: The list is incomplete.*
```
Comment (bittlingmayer) [inline quality/metrics/bleu.md:27]: > Because BLEU is not an exactly defined metric, BLEU scores are not comparable.
> 
> So researchers have created *flavours* that are more concretely defined.
> 
> Here are some of the most well-known flavours of BLEU:

But I'm also fine with just having less :-D

Comment (bittlingmayer) [reply quality/metrics/bleu.md:27]: Maybe this belongs in the top-level metrics comment, since the idea of flavours is true for many metrics.

Maybe "flavours" is too slangy a word.  "variants"?

File: quality/metrics/chrF.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+---
+grand_parent: Building and research
+parent: Metrics
+nav_order: 5
+title: chrF
+description: CHaRacter-level F-score
+---
+
+**chrF** (CHaRacter-level F-score) was introduced in 2015 by Maja Popovic.
```
Comment (bittlingmayer) [inline quality/metrics/chrF.md:9]: Maja Popović

File: quality/metrics/bertscore.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+---
+grand_parent: Building and research
+parent: Metrics
+nav_order: 7
+title: BERTScore
+description:
```
Comment (bittlingmayer) [inline quality/metrics/bertscore.md:6]: Let's add one?

File: quality/metrics/bertscore.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+---
+
+The **BERTScore** metric calculates the similarity in meaning between the machine translation output and a reference.
+BERTScore uses contextual embeddings.
+
+> BERTScore addresses two common pitfalls in n-gram-based metrics. [...] First, such methods often fail to robustly match paraphrases. [...] Second, n-gram models fail to capture distant dependencies and penalize semantically-critical ordering changes For example, given a small window of size two, BLEU will only mildly penalize swapping of cause and effect clauses (e.g. A because B instead of B because A), especially when the arguments A and B are long phrases. In contrast, contextualized embeddings are trained to effectively capture distant dependencies and ordering.
```
Comment (bittlingmayer) [inline quality/metrics/bertscore.md:12]: Let's add an empty line between each sentence in such quotes, even though it wasn't in the original.

File: quality/metrics/bertscore.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
+---
+grand_parent: Building and research
+parent: Metrics
+nav_order: 7
+title: BERTScore
+description:
+---
+
+The **BERTScore** metric calculates the similarity in meaning between the machine translation output and a reference.
```
Comment (bittlingmayer) [inline quality/metrics/bertscore.md:9]: To me, this makes it sound too good.  That's more like what it tries to do.

> The **BERTScore** metric calculates the similarity of the sentence representation of the machine translation output and of a human reference translation.
>
> BERTScore was invented as an improvement on n-gram-based metrics like BLEU.

File: quality/metrics/bertscore.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+BERTScore uses contextual embeddings.
```
Comment (bittlingmayer) [inline quality/metrics/bertscore.md:10]: I'd remove this.  People who know how BERT works don't need it, people who don't know how BERT works won't know after reading this.

The place to explain a bit would be in or after this sentence further down:
> The BERTScore metric uses BERT sentence representation.

**Comment by bittlingmayer** [comment]
We need to somehow add references to quotes like the one about BERTScore, so people can click through, and we need to do it in some consistently formatted way.

I remember @zouharvi has some ideas about references.

File: quality/metrics/nist.md [hunk 9]
```diff
@@ -0,0 +1,18 @@
+---
+grand_parent: Building and research
+parent: Metrics
+nav_order: 6
+title: NIST
```
Comment (bittlingmayer) [inline quality/metrics/nist.md:5]: It's terrible that they named it that.

File: quality/metrics/chrF.md (hunk 4 above)
Comment (bittlingmayer) [reply quality/metrics/chrF.md:9]: The first sentence should probably introduce what's salient about the metric.

In this case, I think that it's character-based.

> **chrF** is a metric for machine translation evaluation that calculates the similarity between a machine translation output and a reference translation using character n-grams, not word n-grams.

Comment (bittlingmayer) [reply quality/metrics/chrF.md:9]: >
> Metrics based on word n-grams are especially problematic for high-morphology languages.
>

File: quality/metrics/chrF.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+chrF is a string-based metric.
```
Comment (bittlingmayer) [inline quality/metrics/chrF.md:10]: I honestly don't know what this means.

File: quality/metrics/comet.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+title: COMET
+description: Crosslingual Optimized Metric for Evaluation of Translation
+---
+
+**COMET** (Crosslingual Optimized Metric for Evaluation of Translation) is a metric based on token or sentence embeddings.
+It is based on similarity of vector representations.
+
+> Traditionally only QE models have made use of the source input, whereas MT evaluation metrics rely instead on the reference translation. [...], we show that using a multilingual embedding space allows us to leverage information from all three inputs and demonstrate the value added by the source as input to our MT evaluation models.
+
+COMET is trained on predicting different types of human judgements in the form of post-editing effort, direct assessment or translation error analysis.
```
Comment (bittlingmayer) [inline quality/metrics/comet.md:14]: > Unlike [BERTScore](/bertscore.md), ...

File: quality/metrics/metrics.md [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+
+- BLEU
+- METEOR
+- NIST
+- chrF
+- TER
+
+*Note: The list is incomplete.*
+
+### Pre-trained metrics
```
Comment (bittlingmayer) [inline quality/metrics/metrics.md:31]: I'd call these "Machine learning metrics", "Machine learning-based metrics",  "Predictive metrics", "Model-based metrics" or something else.

"Pre-trained" is too narrow here, because COMET, for example, is also fine-tuned.

File: quality/metrics/metrics.md [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+- COMET
+- YiSi
+- BERTscore
+
+*Note: The list is incomplete.*
+
+| String-based metrics | Pre-trained metrics |
+| --- | --- |
+| Support any language | Support only pre-trained language |
+| Need reference | Do not reference |
```
Comment (bittlingmayer) [inline quality/metrics/metrics.md:44]: It's not true.

File: quality/metrics/metrics.md [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+| Low correlation with human evaluation | High correlation with human evaluation |
```
Comment (bittlingmayer) [inline quality/metrics/metrics.md:45]: BERT has high correlation when systems are very bad.

As much as I love tables, I'd probably not do this as a table.

I'd do it like this

> ### Language support
> String-based metrics can support any language.  But the density of words and characters vary by language.  So the accuracy varies by language pair, and the results cannot be compared across language pairs.
>
> Machine learning-based metrics require a model that was trained on data with the source and target languages.
> 
> ### Accuracy
> Machine learning-based metrics have a higher correlation with human evaluation.
> 
> But metrics based on machine learning are not explainable or fair, so they cannot be used in a research competition.

File: quality/metrics/metrics.md [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+
+### Metrics timeline
```
Comment (bittlingmayer) [inline quality/metrics/metrics.md:47]: I think we should always avoid using the name of an article in a title

So just 

> ### Timeline

However, I'd call it *History* or *Evolution*.

File: quality/metrics/metrics.md [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+
+String-based metrics measure the lexical matching between the system output and the reference translation.
+
+### Challenges
+
+String-based metrics are challenging because there are many possible translations for a single source text.
+
+### Examples
+
+- BLEU
```
Comment (bittlingmayer) [inline quality/metrics/metrics.md:23]: Hyperlinks :-D

File: quality/metrics/metrics.md (hunk 12 above)
Comment (bittlingmayer) [reply quality/metrics/metrics.md:31]: Also, this should be ## not ###

**Comment by bittlingmayer** [comment]
I think every metric article can start with the same basic sentence format.

> ___ is a metric for automatic evaluation of machine translation that calculates the similarity between a machine translation output and a reference translation using ___.

And the two things in ____ should be bold.

File: quality/metrics/nist.md [hunk 17]
```diff
@@ -0,0 +1,18 @@
...
+description: National Institute of Standards and Technology
+---
+
+**NIST** (National Institute of Standards and Technology) is a string-based metric.
```
Comment (bittlingmayer) [inline quality/metrics/nist.md:9]: I think the part in parentheses should be bold.

**Comment by bittlingmayer** [comment]
Great job keeping these short and well-formatted, the organization and scope are also good.

File: quality/metrics/bleu.md [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+BLEU is an [n-gram](/../concepts/n-gram.md) word-based metric.
+
+BLEU is based on translation precision.
+Its basic unit of evaluation is the sentence.
+
+BLEU treats all n-grams equally.
+
+### Flavours
+
+- sacreBLEU
```
Comment (bittlingmayer) [inline quality/metrics/bleu.md:23]: I would say that sacreBLEU is not a metric, it is an implementation of a few metrics: BLEU, chrF and TER.

**Comment by zouharvi** [comment]
> We need to somehow add references to quotes like the one about BERTScore, so people can click through, and we need to do it in some consistently formatted way.

You mean links to papers, like citations? We could use the Wikipedia citations system<sup>[1](https://en.wikipedia.org/wiki/Wikipedia:Basic_citation_concepts)</sup> and possibly add reference to the footnotes.

1. [en.wikipedia.org/wiki/Wikipedia:Basic_citation_concepts](https://en.wikipedia.org/wiki/Wikipedia:Basic_citation_concepts)

File: quality/metrics/human-evaluation-metrics.md [hunk 19]
```diff
@@ -0,0 +1,92 @@
...
+nav_order: 1
+title: Human evaluation metrics
+description: Metrics for machine translation human evaluation
+---
+
+**Human evaluation** of machine translation is the task of manually assessing the machine translation output following certain criteria.
+
+## Challenges
+
+- Human evaluation is subjective by nature
```
Comment (bittlingmayer) [inline quality/metrics/human-evaluation-metrics.md:13]: These should end up with full-stops.

File: quality/metrics/human-evaluation-metrics.md [hunk 20]
```diff
@@ -0,0 +1,92 @@
...
+- Human evaluation is expensive and time-consuming
```
Comment (bittlingmayer) [inline quality/metrics/human-evaluation-metrics.md:14]: "slow and expensive" is a simpler way to say it.

File: quality/metrics/human-evaluation-metrics.md [hunk 21]
```diff
@@ -0,0 +1,92 @@
...
+- There are no standardised approaches to human evaluation
```
Comment (bittlingmayer) [inline quality/metrics/human-evaluation-metrics.md:15]: Is it really true?

File: quality/metrics/human-evaluation-metrics.md [hunk 22]
```diff
@@ -0,0 +1,92 @@
+---
+grand_parent: Building and research
+parent: Metrics
+nav_order: 1
+title: Human evaluation metrics
+description: Metrics for machine translation human evaluation
```
Comment (bittlingmayer) [inline quality/metrics/human-evaluation-metrics.md:6]: Metrics for human evaluation of machine translation

File: quality/metrics/human-evaluation-metrics.md [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+---
+
+**Human evaluation** of machine translation is the task of manually assessing the machine translation output following certain criteria.
```
Comment (bittlingmayer) [inline quality/metrics/human-evaluation-metrics.md:9]: The bold part should almost always be the same as the title.

> **Human evaluation metrics** for machine translation ...

File: quality/metrics/bertscore.md [hunk 24]
```diff
@@ -20,14 +20,15 @@ BERTScore was invented as an improvement on [n-gram](/../concepts/n-gram.md)-bas
...
 
 The BERTScore metric uses sentence representations from BERT, a deep learning model.
 
 > BERTScore computes precision, recall, and F1 measure.
 >
-> [*Metric: bert_score*](https://huggingface.co/spaces/evaluate-metric/bertscore)
+> [*Metric: bert_score*](#resources-and-papers)
 
-### See more:
+### Resources and papers
```
Comment (bittlingmayer) [inline quality/metrics/bertscore.md:31]: Papers are resources, so just *Resources* is fine.

File: quality/metrics/human-evaluation-metrics.md [hunk 25]
```diff
@@ -0,0 +1,92 @@
...
+- Human evaluation is expensive and time-consuming
+- There are no standardised approaches to human evaluation
+
+## Metrics
+
+### MQM
+
+**Multidimensional Quality Metrics** (**MQM**) is a framework that determines specific translation errors, severities, and error weights.
+
+#### Error dimensions
```
Comment (bittlingmayer) [inline quality/metrics/human-evaluation-metrics.md:23]: I'd probably just make this not a title:

> Error dimensions:

**Review by bittlingmayer — APPROVED** [review]
